### PR TITLE
FEAT: add cc addresses and post_id to sent email logs

### PR DESCRIPTION
**Pull request by tyb-talks**
### Context
Our email logs for `Sent` emails were missing some information in the admin interface that helps with troubleshooting:
* group_smtp emails did not have CC addresses
* post_id was not in the column that had post details

With the additional detail on CC addresses, there's also a need to include that DB column when filtering logs with an email address. As we are only adding CC addresses for group_smtp emails, I've aligned the filter logic with that constraint for now.

Other changes in this PR:
* added slice helper
* truncate addresses into a tooltip once total number of addresses in column is > 3
* shift addresses that match the `address` filter value up to the top of the addresses list, when searching group_smtp emails

### Considerations
The CC addresses could have been added via another column but I figured this made it easier for the filter logic to work (requiring only a single user input to search across `to_address` / `cc_addresses`). Having both types of recipient address in a single cell is also an existing pattern for the `Received` admin email logs tab.

### How to Test
1. Create some sent emails
2. Log in as admin, and navigate to `/admin/email/sent`.
3. Filter with Email Type `group_smtp`.

Screenshot 2023-12-22 at 6 38 49 PM

Before truncating:
before-truncate

With truncated address list, without filter:
truncated-no-filter

With truncated address list, with filters:
truncated-with-address-filter

### TODO
- [x] more tests for added controller & helper logic
- [ ] styling changes if needed

### CODE REVIEW

File: app/assets/javascripts/admin/addon/templates/email-sent.hbs [hunk 1]
```diff
@@ -54,16 +54,29 @@
...
-            <a href="mailto:{{l.to_address}}">{{l.to_address}}</a>
+            <p><a
+                href="mailto:{{l.to_address}}"
+                title="TO"
+              >{{l.to_address}}</a></p>
+            {{#if l.cc_addresses}}
+              {{#each l.cc_addresses as |cc|}}
+                <p><a href="mailto:{{cc}}" title="CC">{{cc}}</a></p>
+              {{/each}}
+            {{/if}}
```
Comment (nattsw) [inline app/assets/javascripts/admin/addon/templates/email-sent.hbs:65]: For better UX, could you do a simple "if there is > 5, truncate into a tooltip?". You can consider using floatkit's tooltip.

Comment (nattsw) [reply app/assets/javascripts/admin/addon/templates/email-sent.hbs:65]: Hmm. I do wonder if it makes sense to put all CC in a tooltip regardless -- will let you decide.

File: app/assets/javascripts/discourse/app/helpers/slice.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { registerRawHelper } from "discourse-common/lib/helpers";
+
+registerRawHelper("slice", slice);
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/helpers/slice.js:3]: Let's drop this line - we don't need this helper to be accessible in `.hbr` files

File: app/assets/javascripts/admin/addon/templates/email-sent.hbs [hunk 3]
```diff
@@ -54,16 +54,67 @@
...
+                  <p><a href="mailto:{{cc}}" title="CC">{{cc}}</a></p>
+                {{/each}}
+                <DTooltip
+                  @placement="right-start"
+                  @arrow={{true}}
+                  @identifier="email-log-cc-addresses"
+                  @interactive={{true}}
+                >
+                  <:trigger>
+                    [See more...]
```
Comment (davidtaylorhq) [inline app/assets/javascripts/admin/addon/templates/email-sent.hbs:80]: I think this should be an I18n string?

File: app/assets/javascripts/admin/addon/templates/email-sent.hbs [hunk 4]
```diff
@@ -54,16 +54,67 @@
...
             <span title={{l.reply_key}} class="reply-key">{{l.reply_key}}</span>
           </td>
           <td class="sent-email-post-link-with-smtp-response">
             {{#if l.post_url}}
-              <a href={{l.post_url}}>{{l.post_description}}</a><br />
-            {{/if}}
+              <a href={{l.post_url}}>
+                {{l.post_description}}
+              </a>
+              (post_id:
```
Comment (davidtaylorhq) [inline app/assets/javascripts/admin/addon/templates/email-sent.hbs:114]: I18n string?

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**